closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityConfigurator.java

```diff
@@ -0,0 +1,322 @@
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+package openj9.internal.security;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import sun.security.util.Debug;
+
+/**
+ * Configures the security providers when in restricted security mode.
+ */
+public final class RestrictedSecurityConfigurator {
+
+    private static final Debug debug = Debug.getInstance("semerufips");
+
+    // Restricted security mode enable check, only supported on Linux x64.
+    private static final boolean userEnabledSecurity;
+    private static final boolean isSecuritySupported;
+    private static final boolean shouldEnableSecurity;
+    private static final String userSecuritySetting;
+    private static final boolean userEnabledFIPS;
+
+    private static boolean loadedProps = false;
+
+    private static int userSecurityNum = 0;
+    private static boolean userSecurityTrace;
+    private static boolean userSecurityAudit;
+    private static boolean userSecurityHelp;
+
+    private static final List<String> supportPlatforms = List.of("amd64");
+
+    static {
+        String[] props = AccessController.doPrivileged(
+                new PrivilegedAction<>() {
+                    @Override
+                    public String[] run() {
+                        return new String[] { System.getProperty("semeru.fips"),
+                                System.getProperty("semeru.restrictedsecurity"),
+                                System.getProperty("os.name"),
+                                System.getProperty("os.arch") };
+                    }
+                });
+        userEnabledFIPS = Boolean.parseBoolean(props[0]);
+        // If semeru.fips is true, then ignore semeru.restrictedsecurity, use userSecurityNum 1.
+        userSecuritySetting = userEnabledFIPS ? "1" : props[1];
+        userEnabledSecurity = !isNullOrBlank(userSecuritySetting);
+        isSecuritySupported = "Linux".equalsIgnoreCase(props[2])
+                && supportPlatforms.contains(props[3]);
+        shouldEnableSecurity = (userEnabledFIPS || userEnabledSecurity) && isSecuritySupported;
```

Comment:
I don't think `userEnabledFIPS` needs to be a field, since if `userEnabledFIPS` is `true` then `userEnabledSecurity` will definitely be `true`, because `userSecuritySetting` is set to `1`. This line can be
```
shouldEnableSecurity = userEnabledSecurity && isSecuritySupported;
```
and line 108 can be
```
if (userEnabledSecurity && !isSecuritySupported) {
```

Reply:
@ZainabF92
 
Hi Zainab,

The flag "userEnabledFIPS" is for back compatibility to the last version of FIPS mode code, which we released at this July release and also published the user guide. So, we have to keep this flag there even it doesn't needed for the restricted security mode.